src/test/java/io/confluent/examples/streams/kafka/EmbeddedSingleNodeKafkaCluster.java

```diff
@@ -43,7 +43,7 @@
 public class EmbeddedSingleNodeKafkaCluster extends ExternalResource {
 
   private static final Logger log = LoggerFactory.getLogger(EmbeddedSingleNodeKafkaCluster.class);
-  private static final int DEFAULT_BROKER_PORT = 0; // 0 results in a random port being selected
+  private static final int DEFAULT_BROKER_PORT = 1234; // pick a random port
```

Comment:
I had some trouble getting this to run locally without setting an explicit port, rather than the current value of `DEFAULT_BROKER_PORT` which is `0`. According to the comment on that line this results in a random port being selected, but in my experience it was actually just selecting the literal `0`. So, I'm picking a random port here. If anyone knows how the "0 --> random port" thing is supposed to work, and why it now isn't, I can look into fixing that up. But as it is I don't really know where to go from here and thus we're just going with `1234`

Reply:
That's probably fine. I don't know how/whether the randomization thing is supposed to work either.

The only concern would be the likelihood of colliding with another process already running on port 1234, which would result in the test failing to bind the port.

From the looks of it, port 1234 is used by VLC and maybe a Monopoly game server? (https://duckduckgo.com/?t=ffab&q=port+1234), so I think we're fine.